### PR TITLE
Pin jupyter-notebook version

### PR DESCRIPTION
**Pull request by danielhollas**
In https://github.com/aiidalab/aiidalab-qe/issues/640, @superstar54 and I were debugging an issue with a stalled pip installation, likely caused by his quite complex dev environment. One thing I noticed that while the pip was trying hard to solve the dependencies, it started backtracking and looking at older versions of some of the core Jupyter libraries, most notably Jupyter notebook (`notebook` package). This is definitely not desirable, and if pip was able to finish, it would likely break the container. 

We've been having discussions in the past about pinning some jupyter-related packages to prevent this scenario from happening. I'll open a separate issue for that, but for starters I think we should pin at least the notebook package. We definitely don't want users to be able to downgrade the version, and we also don't want them to upgrade it, since we know that v7 will currently not work as it is a completely different architecture.

### CODE REVIEW

File: stack/base/Dockerfile
Comment (danielhollas) [inline stack/base/Dockerfile:40]: I've rearranged things a bit so that we define the pinned packages first (both for pip and conda) and only then we do any installation.

File: stack/base/Dockerfile
Comment (danielhollas) [inline stack/base/Dockerfile:47]: pip is originally installed via conda so I've changed this and use `mamba update pip` instead.

Comment (danielhollas) [reply stack/base/Dockerfile:47]: Perhaps we should think about updating conda and mamba as well?

**Comment by danielhollas** [comment]
@unkcpz would you have time to take a look in the next few days? I'd like to publish a new version of the image after we merge this.

**Comment by unkcpz** [comment]
Yes, it is in my list, I'll have a look asap. Sorry.

**Comment by danielhollas** [comment]
No worries at all, thanks!

File: stack/base/Dockerfile
Comment (unkcpz) [reply stack/base/Dockerfile:40]: Make sense!

File: stack/base/Dockerfile
Comment (unkcpz) [reply stack/base/Dockerfile:47]: The mamba is from jupyter image, so I think it is safe to use any version it used, and if I remember correctly they spin up the build process periodically and will try to bring the latest mamba.

**Review by unkcpz — APPROVED** [review]
Looks like a safe move. Thanks @danielhollas.

**Comment by danielhollas** [comment]
Thanks @unkcpz. I've added some tests. After merging this I'm going to publish a new version.

File: stack/base/Dockerfile
Comment (unkcpz) [reply stack/base/Dockerfile:47]: I didn't use aiidalab mamba for quite a while, did you see some annoying warnings ask to update? If so I am totally fine with updating to the latest.

**Review by unkcpz — APPROVED** [review]

File: stack/base/Dockerfile
Comment (danielhollas) [reply stack/base/Dockerfile:47]: I did not, I just think it is a good practice to keep it updated. I'll take a closer look and prepare a PR at some point.

btw: Thanks for the review, I've added some tests in case you want to take a look. Waiting for the build to finish and will then merge.

**Pull request merged**
